package/network/config/ltq-vdsl-app/files/10_atm_ptm.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+[ "$DSL_NOTIFICATION_TYPE" = "DSL_STATUS" ] || exit 0
+
+if [ "$DSL_TC_LAYER_STATUS" = "ATM" ]; then
+
+	if grep -q "ltq_ptm_vr9" /proc/modules ; then
+		# TODO: make ltq_atm_vr9/ltq_ptm_vr9 "hot-swappable"
+		# ifconfig ptm0 down &>/dev/null
+		# rmmod "ltq_ptm_vr9"
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer ATM while PTM driver is loaded. Rebooting System"
+		reboot
+		exit
+	fi
+
+	if ! grep -q "ltq_atm_vr9" /proc/modules ; then
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer ATM"
+
+		modprobe ltq_atm_vr9
+		esi $(printf '%012X' $((1+0x$(tr -d : </sys/class/net/eth0/address))))
+		
+		grep -q "br2684" /proc/modules || modprobe br2684
+		/etc/init.d/br2684ctl start &>/dev/null
+		ip link set dev nas0 up
+	fi
+
+elif [ "$DSL_TC_LAYER_STATUS" = "EFM" ]; then
+	if grep -q "br2684" /proc/modules ; then
+		ifconfig nas0 down &>/dev/null
+		/etc/init.d/br2684ctl stop &>/dev/null
+		rmmod "br2684"
+	fi
+
+	if grep -q "ltq_atm_vr9" /proc/modules ; then
+		# TODO: make ltq_atm_vr9/ltq_ptm_vr9 "hot-swappable"
+		# rmmod "ltq_atm_vr9"
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer EFM/PTM while ATM driver is loaded. Rebooting System"
+		reboot
+		exit
+	fi
+
+	if ! grep -q "ltq_ptm_vr9" /proc/modules ; then
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer EFM/PTM"
+
+		modprobe ltq_ptm_vr9
+		ip link set dev ptm0 down
+		ip link set dev ptm0 address $(printf '%012X' $((1+0x$(tr -d : </sys/class/net/eth0/address))))
```

Comment:
Yes you are right. If you use ucidef_set_interface_macaddr() to specifiy the mac addresses than this code would be unneccessary or probably ends in an different mac.

In my current situation I don't want to use the ucidef_set_interface_macaddr() because this ends up with entries of the mac in /etc/config/network. But in some scenarios you want to be able to clone devices by simply loading the same config-backup from one device into other devices. Doing this with the mac in /e/c/network will result in having a lot of devices with the same macs.
So I prefer loading/calculating the correct mac addresses from the stored base mac address (which is stored in uboot-env in my case) at runtime.

I've also integrated this based on the esi command in the atm section witch is taken from the vdsl_cpe_control_wrapper. Isn't this also faulty then?

We can drop this (and consequently even the esi stuff ?) if you think so.

Any suggestions are welcome.